osh/builtin_lib.py

```diff
@@ -83,7 +84,12 @@ def Run(self, cmd_val):
       return 0
 
     if arg.r:
-      readline.read_history_file(self.GetHistoryFilename())
+      history_filename = self.GetHistoryFilename()
+      history_file_exists = path_stat.exists(history_filename)
+      if not history_file_exists:
+        raise error.ErrExit(1, "The file '%s' ($HISTFILE) does not exist" % history_filename, loc.Missing())
```

Comment:
hm actually `ErrExit` is intended for use with `set -o errexit`

I think simply `return 1` will do what you want here?

Also I would just say `if not path_stat.exists()` instead of the extra var

---

Is this intended to be friendlier than bash?  I think bash just ignores the error?

I tried `bash -i -c 'HISTFILE=a; history -r`

Reply:
Ok. I still want to provide some information about the failure. I see that we use a `self.errfmt.Print_` pattern in other builtins. Would that be appropriate here?

> Is this intended to be friendlier than bash? I think bash just ignores the error?
>
> I tried bash -i -c 'HISTFILE=a; history -r

I do get an error code on my machine (no message though)

```sh
$ HISTFILE=_tmp/dne
$ history -r
$ echo $?
1
```